packages/shared/src/forms/useCalculateSliderPositions.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import * as React from 'react';
+
+type Props = {|
+  +labelEndWidth: number,
+  +labelStartAtMax: boolean,
+  +labelStartWidth: number,
+  +paddingLeft: number,
+  +paddingRight: number,
+  +props: any,
+  +width: number,
+  +setPaddingLeft: number => void,
+  +setLabelStartAtMax: boolean => void,
+  +setPaddingRight: number => void,
+  +min: number,
+  +max: number,
+  +endValue: ?number,
+  +startValue: number,
+|};
+
+export default function useCalculateSliderPositions({
+  labelEndWidth,
+  labelStartAtMax,
+  labelStartWidth,
+  paddingLeft,
+  paddingRight,
+  props,
+  width,
+  setPaddingLeft,
+  setLabelStartAtMax,
+  setPaddingRight,
+  min,
+  max,
+  endValue,
+  startValue,
+}: Props) {
+  React.useEffect(() => {
+    function getOffset(input: number) {
+      let val;
+
+      if (input > max) {
+        val = max;
+      } else if (input < min) {
+        val = min;
+      } else {
+        val = input;
+      }
+      return Math.round((val / (max - min)) * width);
```

Comment:
I believe this could be larger than the width:
max=10
min=1
input = 10 => val = 10

10/9*width > 1*width

Is that the intended behaviour?

Reply:
Maybe it should be `(val - min) / (max - min) * width`, in which case:
* val = min => 0 (no offset)
* val = max => `width` (full offset)
* min < val < max => % of width

Reply:
This gives me a headache 🙈  I mostly copy pasted this from what it was, and it works as intended